gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -286,51 +237,231 @@ public EntityFetcherResponse getAggregatedMetrics(
             i < chunk.getResultSetMetadata().getColumnMetadataCount();
             i++) {
           ColumnMetadata metadata = chunk.getResultSetMetadata().getColumnMetadata(i);
+          org.hypertrace.core.query.service.api.Value columnValue = row.getColumn(i);
+          addAggregateMetric(entityBuilder,
+              requestContext,
+              entitiesRequest,
+              metadata,
+              columnValue,
+              attributeMetadataMap);
+        }
+      }
+    }
+    return new EntityFetcherResponse(entityMap);
+  }
+
+  @Override
+  public EntityFetcherResponse getEntitiesAndAggregatedMetrics(
```

Comment:
@inespo 
I'm trying to understand the need for this method. How is it different from getAggregatedMetrics?

Reply:
So I think I can echo @inespo thoughts when I say that it may not be that different and could actually be merged into it. But this differentiates the entry points for single source and multiple sources. This whole class needs to be refactored and cleaned up.